File: hw90/src/main/java/com/example/hw90/ContactDao.java
package com.example.hw90;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface ContactDao {

    @Insert
    void insert(Contact contact);

    @Update
    void update(Contact contact);

    @Delete
    void delete(Contact contact);

    @Query("SELECT * FROM ContactTable")
    LiveData<List<Contact>> getAllContacts();

    @Query("SELECT * FROM ContactTable WHERE line1 LIKE :filterQuery")
    List<Contact> getFilteredContact(String filterQuery);
}

File: hw5/src/main/java/com/example/hw5/ContactDao.kt
package com.example.hw5

import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface ContactDao {
    @Insert
    fun insert(contact: Contact)

    @Update
    fun update(contact: Contact)

    @Delete
    fun delete(contact: Contact)

    @Query("SELECT * FROM ContactTable")
    fun getAllContacts(): LiveData<List<Contact>>
}